src/client.rs

```diff
@@ -93,6 +93,41 @@ pub struct AppendOptions<'a> {
     pub date: Option<DateTime<FixedOffset>>,
 }
 
+/// A builder for the append command
+#[derive(Default)]
+pub struct AppendCmd<'a> {
+    flags: Vec<&'a Flag<'a>>,
+    date: Option<DateTime<FixedOffset>>,
+}
+
+impl<'a> AppendCmd<'a> {
+    /// Create a new AppendCmd builder
+    pub fn create() -> Self {
+        Self::default()
+    }
+
+    /// Append a flag
+    pub fn flag(&mut self, flag: &'a Flag<'a>) -> &mut Self {
+        self.flags.push(flag);
+        self
+    }
+
+    /// Set the internal date
+    pub fn internal_date(&mut self, date: DateTime<FixedOffset>) -> &mut Self {
+        self.date = Some(date);
+        self
+    }
+}
+
+impl<'a> Into<AppendOptions<'a>> for AppendCmd<'a> {
+    fn into(self) -> AppendOptions<'a> {
+        AppendOptions {
+            flags: Some(&self.flags[..]),
+            date: self.date,
+        }
+    }
+}
```

Comment:
The PR adds both AppendOptions + AppendCmd. Leaving out the latter things are mostly ready. Tests are green and all.

I'm having an issue with AppendCmd and lifetimes with regards to the collected flags:

![image](https://user-images.githubusercontent.com/247048/102382174-1833e380-4005-11eb-97db-a70ba359cd4b.png)

I'm still super comfortable with lifetimes in Rust. Perhaps you can give me some pointers @jonhoo ? I choose to use a Vec to store the flags for the ease of `push()`ing.

I tried adding separate lifespans `'a, 'b: 'a` but that quickly became a mess.